### ISSUE TITLE
Release GMT 6.2.0rc1

### ISSUE THREAD

**Issue by meghanrjones** [GenericMappingTools/gmt#5070]
**Version**:  6.2.0rc1

**Before release**:

- [x] check if all tests pass on macOS, Linux and Windows
- [x] check if other GMT-derived projects work well
	- [x] MB-System (@PaulWessel)
	- [x] GMTSAR (@PaulWessel)
	- [x] GMT.jl (@joa-quim)
	- [x] PyGMT (@leouieda, @seisman, @weiji14)
	- [x] gmtmex (@PaulWessel, @joa-quim)
- [x] run `src/gmt_make_*.sh` to update some .c and .h files
- [x] run `admin/gs_check.sh` to test if latest ghostscript version works
- [x] update [changelog](/doc/rst/source/changes.rst)
- [x] update INSTALL.md
- [x] check if there are any warnings when build the documentation
- [x] add one new entry in `doc/rst/_static/version_switch.js` if it's a minor release
- [x] check/set values in `cmake/ConfigDefault.cmake`
    - [x] `GMT_VERSION_YEAR` is current year
    - [x] `GMT_PACKAGE_VERSION_*` is correctly set
    - [x] `GMT_LIB_SOVERSION` is correctly set
    - [x] set `GMT_PACKAGE_VERSION_SUFFIX` to `rc1`
    - [x] set `GMT_PUBLIC_RELEASE` to `TRUE`
- [x] freeze codes and commit all changes to GitHub

**Release**:

- [x] create source tarballs (tar.gz and tar.xz) (@PaulWessel)
- [x] create macOS bundle (@PaulWessel)
- [x] create Windows installers (win64) (@joa-quim)
- [x] check if the source tarballs, macOS bundle and Windows installers work well
- [x] upload source tarballs, macOS bundle, Windows installers to the GMT FTP (@PaulWessel)
- [x] make a tag and push it to github (**Must be done after uploading packages to the GMT FTP**)
    ```bash
    # checkout master (for minor releases) or 6.x branch (for patch releases)
    git checkout XXXX
    # create the tag x.x.x
    git tag x.x.x
    # Push tags to GitHub
    git push --tags
    ```
- [x] make a GitHub release.
  The GitHub Actions automatically create a draft release after pushing the tag to github.
  We need to go to the [GitHub Release](https://github.com/GenericMappingTools/gmt/releases) page, and review it manually.
  - [x] 5 files are attached as release assets (2 source tarballs, 2 installers and 1 checksum file).
  - [x] download the checksum file and check if the checksums are correct
  - [x] edit the draft release, set the target to the correct tag, and publish the release
- [x] make announcements in the [GMT forum](https://forum.generic-mapping-tools.org/)
- [x]  update links on the main site (News, Download & Documentation)

**After release**:

- [x] comment the `GMT_PACKAGE_VERSION_SUFFIX` line in `cmake/ConfigDefault.cmake`
- [x] comment the `set (GMT_PUBLIC_RELEASE TRUE)` line
- [x] commit changes to GitHub

**3rd-party update**

- [x] update [conda-forge feedstock](https://github.com/conda-forge/gmt-feedstock) (@leouieda, @seisman, @weiji14)

---

- [x] Party :tada: (don't tick before all other checkboxes are ticked!)

**Comment by meghanrjones** [comment]
I used the full release template for the release candidate. Please edit as necessary for rc1 @PaulWessel, @joa-quim, @seisman.

I will work on the changelog today.

**Comment by seisman** [comment]
FYI, ghostscript just released v9.54.0 two days ago. We may need to run GMT tests using gs 9.54.0 to make sure it doesn't break anything.

**Comment by meghanrjones** [comment]
A couple questions:
1. Are the DOI related steps necessary for release candidates?
2. Will we only upload the tarballs to GitHub or also do the 3rd party updates?

**Comment by seisman** [comment]
> 1. Are the DOI related steps necessary for release candidates?

I think no. It's better to make RC releases as simple as possible.

> 2. Will we only upload the tarballs to GitHub or also do the 3rd party updates?

Most 3rd party package managers don't accept RC or beta releases and some are slow in accepting updates. The only exception is conda-forge.

**Comment by PaulWessel** [comment]
@seisman, is it doable for you to install gs 9.54 via brew and see what it does with the master tests? Not so easy with macports I think.

**Comment by seisman** [comment]
I failed to build gs 9.54 so can't test it. It would be easier to test it for Linux or Windows users, because gs provides binary packages (https://www.ghostscript.com/download/gsdnld.html).

**Comment by PaulWessel** [comment]
We do want DCW 2.0 for this.  Will this be a problem for distros?

**Comment by seisman** [comment]
> We do want DCW 2.0 for this. Will this be a problem for distros?

If I remember it correctly, DCW 2.0 doesn't work for GMT <= 6.1.1, right? We can release dcw 2.0.0, but we need to warn users and package managers that dcw 2.0.0 requires GMT>=6.2.0.

**Comment by PaulWessel** [comment]
Just reminding myself what build-release.sh does and when it calls build-macos-external-list.sh it is not pretty.  I think this is fragile stuff.

**Comment by joa-quim** [comment]
I have no intention to build 9.54 on Windows. Artifex has a very non-recommendable recent past of breaking changes and there was a post after 9.54 release reporting troubles building on OSX.

**Comment by PaulWessel** [comment]
I think 9.53 is fine for the installers but we do want to find out if they break stuff again since outside win and macos installers it will be whatever gs version people have installed, and they will have 9.54 soon.

**Comment by PaulWessel** [comment]
Getting a bit further after updating the ps gs test files.  It builds the list of libraries for the bundle, but then failes with this error:

```
CMake Error at cmake/ConfigUser.cmake:29:
  Parse error.  Expected "(", got newline with text "

  ".
```

When cmake runs it is actully reading the admin/ConfigReleaseBuild.cmake that has been copied over, and its line 29 is this:

`set (EXTRA_INCLUDE_EXES "${CMAKE_BINARY_DIR}/add_macOS_cpack.txt")`

So, it would see CMAKE_BINARY_DIR is empty, @seisman ? BTW, would have been handy to tel cmake to read its ConfigUser.cmake from a specified directory instead of forcing me to override mine...

**Comment by PaulWessel** [comment]
OK, moving along.  The build-release.sh now completes after building tarballs and bundle.  Here are some messages that we perhaps need to look at more carefully:

```
[691/694] Building manpages...
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/index.rst:39: WARNING: Unknown directive type "panels".

.. panels::

    .. toctree::
        :maxdepth: 1
        :caption: Getting started

        gallery
        animations
        tutorial
```

Then a bunch of these:

```
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/animations.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/api.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/changes.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/cookbook.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/datasets.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/debug.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/deprecated-defaults.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/devdocs.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/gallery.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/migrating.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/modules.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/modules-classic.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/postscriptlight.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/rst-cheatsheet.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/std-opts-classic.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/switching.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/tutorial.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/users-contrib-scripts.rst: WARNING: document isn't included in any toctree
/Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/users-contrib-symbols.rst: WARNING: document isn't included in any toctree
```

At the end a pop-up box requiring my authentication came up (because of the specific check on user pwessel in _ConfigReleaseBuild.cmake_), and after that we got this:

`CPack: - Application has been codesigned`

The files then auto-copied themselves to the pwessel SOEST ftp directory we have used for releases.

We will see if this actually means that users can install the bundle without having to fight the Apple gatekeeper.

**Comment by seisman** [comment]
> [691/694] Building manpages...
> /Users/pwessel/GMTdev/gmt-dev/build/doc/rst/source/index.rst:39: WARNING: Unknown directive type "panels".
> 
> .. panels::
> 
>     .. toctree::
>         :maxdepth: 1
>         :caption: Getting started
> 
>         gallery
>         animations
>         tutorial

It's harmless, because the `panels` extension only works for HTML, not manpage.

**Comment by PaulWessel** [comment]
OK, and the other one is probably related since I do not see them when I build the HTML docs.
I am resurrecting the abandoned add-gmtmex-supplement branch after bringing it up to date with master. Still seems like a better way to build the gmtmex rather than old configure scripts.

**Comment by seisman** [comment]
> OK, and the other one is probably related since I do not see them when I build the HTML docs.

Yes, I will see if there is a solution for this when I have time.

**Comment by PaulWessel** [comment]
The building of the GMT/MEX toolbox is moving forward and we probably have to rediscover what we tried 6 months ago or more...  Currently, the gmtmex_parser.c functions are compiled and placed in the libgmt.dylib library:

```
nm -gU libgmt.dylib | grep MEX
00000000002358e0 T _GMTMEX_Get_Object
00000000002327f0 T _GMTMEX_Set_Object
0000000000232720 T _GMTMEX_objecttype
0000000000232700 T _GMTMEX_print_func
```

The gmt.m file is placed in the bin directory with the gmt executable.  What is not yet happening is building the **gmtmex** "executable" or "shared library" that Matlab can load when gmt.h calls it.  We need to add a target of some type and the specify it should be linked with GMT as well as the MATLAB_LIB1 and MATLAB_LIB2 libraries determined by Cmake...

**Comment by meghanrjones** [comment]
Related to https://github.com/GenericMappingTools/gmt/issues/5097#issuecomment-813459344, do others prefer to only add documentation dropdowns for full releases or will a 6.2rc1 dropdown be added to the html docs?

My preference is to add the dropdown option to offer easier access to the documentation for people testing the code and to encourage feedback on the documentation part of the release.

**Comment by PaulWessel** [comment]
> > We do want DCW 2.0 for this. Will this be a problem for distros?
> 
> If I remember it correctly, DCW 2.0 doesn't work for GMT <= 6.1.1, right? We can release dcw 2.0.0, but we need to warn users and package managers that dcw 2.0.0 requires GMT>=6.2.0.

True, but we do want 2.0.0 out since it fixes a problem that required a change in the files.  So maybe we set a 2.0.0 dependency on 6.2.0.  We cannot do much about detecting 2.0.0 from 6.1.1 etc but at least if we require it on our description to packge managers we should be OK [and @joa-quim will add it to his Win installer and we in the bundle].

**Comment by PaulWessel** [comment]
ncdump on 2.0.0 says

```
:title = "DCW-GMT - The Digital Chart of the World for the Generic Mapping Tools" ;
:source = "Processed by the GMT Team, 2020-AUG-01" ;
:version = "2.0.0" ;
:gmtversion = "6.1.1" ;
```

If true it would work with 6.1.1.  We should find out.

**Comment by PaulWessel** [comment]
I have verified that 6.1.1 works with 2.0.0 so I think we should be good.  F'em if they are running something old and decide to just upgrade DCW.

**Comment by meghanrjones** [comment]
I built the documentation - the only warnings were related to the html sources with the manpages target which Paul posted earlier so I checked that box.

**Comment by meghanrjones** [comment]
I checked off off `run admin/gs_check.sh`. Here is the output (using 9.53.3):
```
Convert PS_orig_for_gs.ps to PS_orig_for_gs.pdf in order to achieve transparency
Convert PS_orig_for_gs.pdf to PS_orig_for_gs.png
Compare PS_orig_for_gs.png to ../admin/PS_orig_for_gs.png
Image Difference (RootMeanSquaredError):
           Normalized    Absolute
          ============  ==========
     Red: 0.0000000000        0.0
   Green: 0.0000000000        0.0
    Blue: 0.0000000000        0.0
 Opacity: 0.0000000000        0.0
   Total: 0.0000000000        0.0
```

**Comment by meghanrjones** [comment]
I do not see anything that needs to be updated in `INSTALL.md` so I checked that off as well.

**Comment by weiji14** [comment]
Roughly at what point will there be a 6.2.0rc1 git tag? I've got a conda-forge PR ready at https://github.com/conda-forge/gmt-feedstock/pull/142, and am debating whether to merge that first, or wait for an official 6.2.0rc1 tag (in case any new commits pop up).

**Comment by seisman** [comment]
> Roughly at what point will there be a 6.2.0rc1 git tag? I've got a conda-forge PR ready at [conda-forge/gmt-feedstock#142](https://github.com/conda-forge/gmt-feedstock/pull/142), and am debating whether to merge that first, or wait for an official 6.2.0rc1 tag (in case any new commits pop up).

There are at least one commits (setting `GMT_PUBLIC_RELEASE` to `TRUE`) before making the 6.2.0rc1 tag.

**Comment by seisman** [comment]
Just one more note, since this is a RC release, we also need to set `GMT_PACKAGE_VERSION_SUFFIX` to `rc1` before setting `GMT_PUBLIC_RELEASE` to TRUE.

**Comment by PaulWessel** [comment]
I checked that SOVERSION is set, but checking: I assume it stays at 6?

**Comment by seisman** [comment]
> I checked that SOVERSION is set, but checking: I assume it stays at 6?

Yes.

**Comment by PaulWessel** [comment]
gmtmex finally working after #5116.

**Comment by PaulWessel** [comment]
I have created the tarballs and the macOS bundle:

```
790dbc0cff1de9cf44992163f0c79bec971637a2a6510c5bca1ca74b1ee87cbd  gmt-6.2.0-rc1-darwin-x86_64.dmg
84602244abc245dbcede8ffdf5e340fdcd5c201b246d18486b647ade59a1b4fe  gmt-6.2.0-rc1-src.tar.gz
57039fd21bb95fc76270b859a703a18f63835a4396dd2fae58fd458dbf4c9eb9  gmt-6.2.0-rc1-src.tar.xz
```

For testing, currently at ftp.soest.hawaii.edu://pwessel/release

**Comment by PaulWessel** [comment]
Time to build Win installer and test the installers, per checklist.

**Comment by seisman** [comment]
> I have created the tarballs and the macOS bundle:
> 
> ```
> 790dbc0cff1de9cf44992163f0c79bec971637a2a6510c5bca1ca74b1ee87cbd  gmt-6.2.0-rc1-darwin-x86_64.dmg
> 84602244abc245dbcede8ffdf5e340fdcd5c201b246d18486b647ade59a1b4fe  gmt-6.2.0-rc1-src.tar.gz
> 57039fd21bb95fc76270b859a703a18f63835a4396dd2fae58fd458dbf4c9eb9  gmt-6.2.0-rc1-src.tar.xz
> ```
> 
> For testing, currently at ftp.soest.hawaii.edu://pwessel/release

I thought the tarball names should be like `gmt-6.2.0rc1-xxx`, without the hyphen between 6.2.0 and rc1.

**Comment by PaulWessel** [comment]
Happy to oblige.  I will update PR and rerun.

**Comment by PaulWessel** [comment]
Files updated:

```
3666a48ac16295b552c7d5d9b330a09369c1228129c374d6fd8653d2eade3c35  gmt-6.2.0rc1-darwin-x86_64.dmg
0d0dac3588ac550f152a46814aeff008bc39e3a3f0ce3e7afbeec5ea303f7b38  gmt-6.2.0rc1-src.tar.gz
664bc6c3b322b055c04c1a1c0806819023e90f8a960c547eb42af0f8fb2aa366  gmt-6.2.0rc1-src.tar.xz
```

Pinging @joa-quim for winballs.

**Comment by seisman** [comment]
The source tarballs look good.

For the macOS bundle, it seems gs 9.50 is included. Is it intended?

**Comment by seisman** [comment]
BTW, for the macOS bundle, I no longer see the "unidentified developers" warning in #2087.

**Comment by PaulWessel** [comment]
Snap, I forgot to fix the gs thing.  No, did not mean to do 9.50 - I think when 6.1.1 was done we had some issues with gs having bugs so I installed 9.50 in a different path.  
SO this is the first bundle where I tried to authenticate it with the UH developer codesign so perhaps that actually worked.

**Comment by PaulWessel** [comment]
Switch to macports path for gs etc in the script build-macos-external-list.sh
Built the dmg and running it.  I notice this:

```
pwessel@macnut:/Applications/GMT-6.2.0rc1.app/Contents/Resources/bin-> ls
ffmpeg			gmt			gmtswitch		ogr2ogr
gdal_translate		gmt-config		gsc
gm			gmt_shell_functions.sh	isogmt
```

There is no longer a gs binary, just gsc.  Both of those are in the macports path at /opt/local/bin and build-macos-external-list.sh lists gs as an executable but instead we place gsc.  So I think when I run

`gmt coast -RFR -JM6i -Gred -B -pdf t`

gmt psconvert is calling gs and probalby finding the one in macport path since there is no gs in the bundle.  THoughts?

**Comment by PaulWessel** [comment]
OK, it is because gs is a link for macports:

`lrwxr-xr-x    1 root  admin         3 Jan 18 02:35 gs -> gsc`

and my otools.c program skips symbolic links.  Options:

1. Train psconvert to look for either gs or gsc (since apparently that is the program)
2. Add a gs symbolic link into the bundle.
3. Rename gsc to gs while building the bundle.

Ideas, @seisman ?

**Comment by PaulWessel** [comment]
4. Add another config item to config.h.in:

```
/* Name of gs executable */
#define GMT_GS_EXECUTABLE "@ GMT_GS_EXECUTABLE@"
```

and then use **GMT_GS_EXECUTABLE** instead of "gs" where it appears.  Seems the best solution and we would juse set **GMT_GS_EXECUTABLE** to **gsc** for the bundle building, else default to **gs**.

**Comment by PaulWessel** [comment]
I like the idea of setting **GMT_GS_EXECUTABLE** to default to _gswin64c_ on Windows which is the fallback if _psconvert_ghostbuster_ cannot fish the path from the registry.

**Comment by seisman** [comment]
I still like to see `gs` in the bin directory, since `gs` is the standard name.

**Comment by PaulWessel** [comment]
OK, and for the bundle that is fine, but for the wider ecosystem there is no **gs** for Windows, only those **gswin64** standard names.  So maybe both solutions are OK to implement:

1. A configure part for the standard name of the ghostscript executable [gs on *unix, gswin64 on Windows]
2. Add a patch to the _build-macos-external-list.sh_  that creates the missing link from **gs** to **gsc** [and this may change over time].

**Comment by seisman** [comment]
> OK, and for the bundle that is fine, but for the wider ecosystem there is no **gs** for Windows, only those **gswin64** standard names. So maybe both solutions are OK to implement:
> 
> 1. A configure part for the standard name of the ghostscript executable [gs on *unix, gswin64 on Windows]
> 2. Add a patch to the _build-macos-external-list.sh_  that creates the missing link from **gs** to **gsc** [and this may change over time].

Sounds good to me.

FYI, the Homebrew ghostscritp has both gs and gsc, and neither of them is symlink.
```
$ ls -l gs gsc
-r-xr-xr-x  1 seisman  admin  13611096 Oct  1  2020 gs
-r-xr-xr-x  1 seisman  admin     50232 Apr  9 17:54 gsc
```

**Comment by PaulWessel** [comment]
Yes, brew and ports do things differently, not clear why or how.  I will make a branch for the above.

**Comment by PaulWessel** [comment]
Updated tarballs and bundle after #5123:

```
20fd68063692f66eec8fabe4c663d4681799ecc972729e1d735d5dcbc703b050  gmt-6.2.0rc1-darwin-x86_64.dmg
c4ee3ab9d82e74928bd63de5606206f622cc513db90716009b24866ab1525e04  gmt-6.2.0rc1-src.tar.gz
a0e3a6728fb64f545b165a25d9c8f0c2281c3405dc32b720c3499d96ba98e64a  gmt-6.2.0rc1-src.tar.xz
```

I suggest rc1 will not be universal binary as I need to learn more about how to do this and merge in M1 binaries.

**Comment by seisman** [comment]
Running any GMT plotting command gives me the following error:
```
psconvert [ERROR]: System call [gs -q -dNOSAFER -dNOPAUSE -dBATCH -sDEVICE=bbox -DPSL_no_pagefill -dMaxBitmap=2147483647 -dUseFastColor=true '/Users/seisman/.gmt/sessions/gmt_session.4734/gmt_0.ps-' 2> '/Users/seisman/.gmt/sessions/gmt_session.4734/psconvert_5035c.bb'] returned error 256.
end [ERROR]: Failed to call psconvert
end [ERROR]: gmtinit_process_figures returned error 79
coast [ERROR]: Unable to call module end for a one-liner plot.
```

Runnin `gs` gives me:
```
$ gs
GPL Ghostscript 9.53.3: Can't find initialization file gs_init.ps.
```

**Comment by PaulWessel** [comment]
RIght.  I need to edit the script that is run when you double-click the bundle icon.    Where is that thing?

**Comment by seisman** [comment]
I think you mean https://github.com/GenericMappingTools/gmt/blob/master/cmake/dist/startup_macosx.sh.in.

**Comment by PaulWessel** [comment]
Found it: cmake/dist/startup_macosx.sh.in.

**Comment by PaulWessel** [comment]
Looks like we need to set GS_LIB or some other parameter because gs does not know where things are.

**Comment by PaulWessel** [comment]
Since you must use a clean machine or hiding gs, might you try to see what GS_LIB needs to be? i.e.,  ${BUNDLE_RESOURCES}/share/ghostscript or /share/ghostscript/9.53.3

**Comment by PaulWessel** [comment]
If I hide /opt/local/bin/gs I still can run gmt coast ... -pdf t and get a map.  What else do I need to do to reproduce your non-gs environment?

**Comment by seisman** [comment]
I just uninstalled my Homebrew gs, because it only takes a few seconds to reinstall it.

**Comment by seisman** [comment]
After clicking the application icon, I need to run the following command to use gs:
```
export GS_LIB=${GMTHOME}/share/ghostscript/9.53.3/Resource/Init/
```

**Comment by PaulWessel** [comment]
OK, I am there too and get your failure. ANd see you just posted value.  I will test taht.

**Comment by joa-quim** [comment]
I've made the Win installers already, Posted the link in Slack.

**Comment by PaulWessel** [comment]
As gcc is still not available for Apple Silicon, I can only build an arm64 bundle using Xcode, which means no OPenMP.  Maybe time to tackle OPenMP and Xcode as I know there are ways.

**Comment by meghanrjones** [comment]
Updated after https://github.com/GenericMappingTools/gmt/pull/5125:
```
d6bc8b173d0ebd3206fb0b69fdbd0abbf350f9a49aec98f124e2cba1e0102903  gmt-6.2.0rc1-darwin-x86_64.dmg
e49e7da3590c2f2d4d24b43b7c63df0757e78c1208367423e180641b1472cd45  gmt-6.2.0rc1-src.tar.gz
e1af8a7ff80b7f085300333ad981643a9be88eb28ed91b8f94fc4068196e7c2c  gmt-6.2.0rc1-src.tar.xz
```

**Comment by seisman** [comment]
They all look good to me.

**Comment by meghanrjones** [comment]
It's not clear to me how the draft-release workflow will work since the files are not in the gmt ftp. Do they need to get moved now? Will the workflow still fail because there is no windows 32 bit installer?

**Comment by seisman** [comment]
> It's not clear to me how the draft-release workflow will work since the files are not in the gmt ftp. Do they need to get moved now? Will the workflow still fail because there is no windows 32 bit installer?

I think they should be moved to the GMT FTP.

**Comment by meghanrjones** [comment]
Paul and I decided that it's not necessary to update the ftp README and VERSIONS for rc's. Should we setup the testbot to run off the ftp or push the tags and run the testbot after?

**Comment by meghanrjones** [comment]
Another question: the draft-release worflow expects both win32 and win64 installers, but we have decided to provide only win64 installers for release candidates. Is it better to edit the workflow for rc1 and change it back after or have a separate workflow that is triggered when '[0-9]+.[0-9]+.[0-9]+rc[0-9]+' tags are pushed?

**Comment by seisman** [comment]
> Another question: the draft-release worflow expects both win32 and win64 installers, but we have decided to provide only win64 installers for release candidates. Is it better to edit the workflow for rc1 and change it back after or have a separate workflow that is triggered when '[0-9]+.[0-9]+.[0-9]+rc[0-9]+' tags are pushed?

It's complicated, perhaps the easiest way is to edit the workflow temporarily. We can find a better solution later.

**Comment by meghanrjones** [comment]
The updated tarballs and bundle are on the ftp with these SHA-256 hashes:
```
f18c4f7be5763bec7e23c0d2d68244ca962af6b46592251cb06d9ee9f7b1a6d3  gmt-6.2.0rc1-darwin-x86_64.dmg
6c88615c5a5dcbaca5927ad1c172f65ab720dece7009b311c80e1d75585a0cbd  gmt-6.2.0rc1-src.tar.gz
1ea45a034cf38197e2ee40507a2f3047df5783b614156aa1229a64a38024ef3c  gmt-6.2.0rc1-src.tar.xz
```
Some small tests ran successful on the bundle with the gmt-release-testbot. There seems to be a bug with running movie using .bat files on windows, but more basic tests pass.

**Comment by meghanrjones** [comment]
I will push a 6.2.0rc1 tag in ~30 minutes unless there are any final comments.

**Comment by meghanrjones** [comment]
It seems that this line in draft-release.yml:
```
run: echo "::set-env name=GMT_VERSION::${GITHUB_REF##*/}"
```
needs to be:
```
run: echo "GMT_VERSION=${GITHUB_REF##*/}" >> $GITHUB_ENV
```

Does this look correct @seisman? How do you recommend changing the workflow, given that the tag has already been pushed?

**Comment by seisman** [comment]
I think it's correct. Given that you already made the tag, I think for 6.2.0rc1, you can just manually create a new release and upload the tarballs, macos bundle and windows installer. We can fix it in rc2.

**Comment by meghanrjones** [comment]
Draft release is up. One quick question: do you recommend this option for the rc?
![image](https://user-images.githubusercontent.com/14077947/115313088-a7324300-a140-11eb-8699-fa6091538b94.png)

**Comment by seisman** [comment]
This is something new to me, but it looks good

**Comment by meghanrjones** [comment]
I can make a post on the forum. Do we need to update the website for release candidates? I think that it is not necessary. @weiji14, I think all the prerequisite steps for conda-forge are now complete.

**Comment by weiji14** [comment]
> I can make a post on the forum. Do we need to update the website for release candidates? I think that it is not necessary. @weiji14, I think all the prerequisite steps for conda-forge are now complete.

Ok, I'll open a Pull Request for a GMT 6.2.0rc1 package on conda-forge.

**Comment by weiji14** [comment]
Some questions: Does GMT 6.2.0rc1 work with dcw-gmt 1.1.4? I'm wondering whether we need to update the conda-forge feedstock at https://github.com/conda-forge/dcw-gmt-feedstock too (currently at 1.1.4). Plus, do we need to pin GMT 6.2.0rc1 to require dcw-gmt 2.0.0 or higher?

**Comment by meghanrjones** [comment]
> Some questions: Does GMT 6.2.0rc1 work with dcw-gmt 1.1.4? I'm wondering whether we need to update the conda-forge feedstock at https://github.com/conda-forge/dcw-gmt-feedstock too (currently at 1.1.4). Plus, do we need to pin GMT 6.2.0rc1 to require dcw-gmt 2.0.0 or higher?

I only started working with dcw-gmt 2.0.0 ~1 week ago, so I think it should be fine to work dcw-gmt 1.1.4. dcw-2.0.0 requires GMT>=6.1.1 but I don't think there's a need to pin any dcw requirements for GMT 6.2.0rc1.

**Comment by PaulWessel** [comment]
Will work fine with 1.1.4, but we want to move forward. 6.1.1 was first version that supported 2.0.0 even though we did not release 2.0.0 at that time.

**Comment by weiji14** [comment]
Ok, I won't do a hard pin to dcw-gmt > 2.0.0 then, but we should probably update https://github.com/conda-forge/dcw-gmt-feedstock to 2.0.0 before the official gmt 6.2.0 release so others can install it in the conda world.

P.S. Not that I have a Mac, but I took the liberty to open https://github.com/conda-forge/dcw-gmt-feedstock (following https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/#how-to-add-a-osx-arm64-build-to-a-feedstock) to help a bit with #4442 (Apple Silicon builds).

**Comment by weiji14** [comment]
> P.S. Not that I have a Mac, but I took the liberty to open [conda-forge/conda-forge-pinning-feedstock#1434](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/1434) (following https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/#how-to-add-a-osx-arm64-build-to-a-feedstock) to help a bit with #4442 (Apple Silicon builds).

Ok, https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/1434 is merged, but need to do upstream dependencies too at https://github.com/conda-forge/gshhg-gmt-feedstock/pull/5 and https://github.com/conda-forge/dcw-gmt-feedstock/pull/5. We may need @leouieda's help to approve and merge those PRs since he's the recipe maintainer for those repos.

**Comment by weiji14** [comment]
> Do we need to update the website for release candidates? I think that it is not necessary.

We updated the website before for GMT 6.0.0 RC releases, so I made one at https://github.com/GenericMappingTools/website/pull/105. Will send out a tweet too once that is merged :wink:

**Comment by PaulWessel** [comment]
Thanks to all, especially @meghanrjones who helped get rc1 on the road.
